File: PHP/NumberOfZeroFilledSubarraysTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NumberOfZeroFilledSubarraysTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("NumberOfZeroFilledSubarrays.php");

class NumberOfZeroFilledSubarraysTest extends TestCase {

    public function testNumberOfZeroFilledSubarrays() {
        $obj = new NumberOfZeroFilledSubarrays();
        //Test 1
        $want = 6;
        $got = $obj->zeroFilledSubarray( [1,3,0,0,2,0,0,4] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 9;
        $got = $obj->zeroFilledSubarray( [0,0,0,2,0,0] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/NumberOfZeroFilledSubarrays_test.go
/**
 * @version GO 1.20.2
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestNumberOfZeroFilledSubarrays
 */
package GO

import (
	"reflect"
	"testing"
)

func TestNumberOfZeroFilledSubarrays(t *testing.T) {
	tables := []struct {
		want int64
		nums []int
	}{
		{6, []int{1, 3, 0, 0, 2, 0, 0, 4}},
		{9, []int{0, 0, 0, 2, 0, 0}},
	}

	for _, table := range tables {
		got := zeroFilledSubarray(table.nums)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
